src/odemis/driver/semnidaq.py

```
@@ -2745,6 +2783,67 @@ def _setTranslation(self, value):
+        If None, and scanPath is defined, the TTL output is set low during the entire scanning.
+        :return: the TTL data as-is, if valid
+        :raises ValueError: if the TTL data is not valid
+        """
+        if ttl is None:  # It's fine, just means no vector scanning or no scan TTL
+            va = getattr(self, va_name)
+            if va.value is not None:  # Previous value
+                self._on_setting_changed()
+            return ttl
+
+        if ttl.ndim != 1 or ttl.shape[0] == 0 or ttl.dtype != numpy.bool_:
+            raise ValueError(f"{va_name} should be boolean of shape (N,), but got {ttl.shape}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)